apps/web/src/components/Basenames/UsernameProfileTransferOwnershipModal/index.tsx

```diff
@@ -0,0 +1,188 @@
+import SearchAddressInput from 'apps/web/src/components/SearchAddressInput';
+import { isAddress } from 'viem';
+import { useAccount } from 'wagmi';
+import { useUsernameProfile } from 'apps/web/src/components/Basenames/UsernameProfileContext';
+import { useErrors } from 'apps/web/contexts/Errors';
+import { Icon } from 'apps/web/src/components/Icon/Icon';
+import {
+  OwnershipSteps,
+  useProfileTransferOwnership,
+} from 'apps/web/src/components/Basenames/UsernameProfileTransferOwnershipModal/context';
+import AllOwnershipTransactionsState from 'apps/web/src/components/Basenames/UsernameProfileTransferOwnershipModal/AllOwnershipTransactionsState';
+import WalletIdentity from 'apps/web/src/components/WalletIdentity';
+import BasenameIdentity from 'apps/web/src/components/BasenameIdentity';
+
+const ownershipStepsTitleForDisplay = {
+  [OwnershipSteps.Search]: 'Send name',
+  [OwnershipSteps.OwnershipOverview]: "You'll be sending",
+  [OwnershipSteps.WalletRequests]: 'Confirm transactions',
+  [OwnershipSteps.Success]: '',
+};
+
+type UsernameProfileTransferOwnershipModalProps = {
+  isOpen: boolean;
+  onClose: () => void;
+};
+
+export default function UsernameProfileTransferOwnershipModal({
+  isOpen,
+  onClose,
+}: UsernameProfileTransferOwnershipModalProps) {
+  // Hooks
+  const { address } = useAccount();
+  const { profileOwnerRefetch, setShowProfileSettings, profileUsername } = useUsernameProfile();
+  const { logError } = useErrors();
+  const {
+    isSuccess,
+    currentOwnershipStep,
+    setCurrentOwnershipStep,
+    recipientAddress,
+    setRecipientAddress,
+    ownershipSettings,
+    batchTransactionsEnabled,
+  } = useProfileTransferOwnership();
+
+  // States
+  const isValidRecipientAddress = isAddress(recipientAddress);
+
+  // Callbacks
+  const onChangeSearchAddress = useCallback(
+    (value: string) => {
+      setRecipientAddress(value);
+    },
+    [setRecipientAddress],
+  );
+
+  const onConfirmRecipientAddress = useCallback(() => {
+    setCurrentOwnershipStep(OwnershipSteps.OwnershipOverview);
+  }, [setCurrentOwnershipStep]);
+
+  const onBackClick = useCallback(() => {
+    if (currentOwnershipStep === OwnershipSteps.OwnershipOverview)
+      setCurrentOwnershipStep(OwnershipSteps.Search);
+  }, [currentOwnershipStep, setCurrentOwnershipStep]);
+
+  const onUpdateOwnershipClick = useCallback(() => {
+    if (!address) return;
+    if (!isValidRecipientAddress) return;
+
+    setCurrentOwnershipStep(OwnershipSteps.WalletRequests);
+  }, [address, isValidRecipientAddress, setCurrentOwnershipStep]);
+
+  const handleOnClose = useCallback(() => {
+    if (currentOwnershipStep === OwnershipSteps.Success) {
+      profileOwnerRefetch()
+        .then(() => {
+          setShowProfileSettings(false);
+          onClose();
+        })
+        .catch((error) => {
+          logError(error, 'Failed to refetch Owner');
+        });
+
+      return;
+    }
+
+    onClose();
```

Comment:
nit: reverse the order here so the one-line case is executed first. IMO makes it a bit more readable when simple condition/run/exit stuff happens at the top